src/algebra/lie/non_unital_non_assoc_algebra.lean

```diff
@@ -23,9 +23,16 @@ algebra and we provide some basic definitions for doing so here.
 
 ## Main definitions
 
-  * `lie_ring.to_non_unital_non_assoc_semiring`
+  * `commutator_ring` turns a Lie ring into a `non_unital_non_assoc_semiring` by turning its
+    `has_bracket` (denoted `⁅, ⁆`) into a `has_mul` (denoted `*`).
   * `lie_hom.to_non_unital_alg_hom`
 
+## Implementation notes
+
+Previously the structure was added by a non-instance `lie_ring.to_non_unital_non_assoc_semiring`,
+which could be made an instance locally. This caused instance diamonds since all rings are
```

Comment:
Worse than a diamond - this created a cycle, didn't it?

Reply:
Not quite: the instances go `ring → lie_ring → non_unital_non_assoc_semiring`, so without the associativity there is no cycle.